### PR TITLE
Add theme block type completion in `{% schema %}`

### LINKED ISSUE #627: Theme block type (definition) completion
In a theme or section schema, when a user is completing top level `"blocks.[].type"` values, we should complete all of
- `@theme` when it makes sense
- `@app` when it makes sense
- in section files,
   - if no local block exist
      - theme block names
- in block files
  - theme block names

### PR DESCRIPTION
**Pull request by charlespwd**
## What are you adding in this PR?

The `BlockTypeCompletionProvider` offers value completions of the `blocks.[].type` property inside section and theme block `{% schema %}` tags.

```liquid
{% schema %}
{
  "blocks": [
    { "type": "█" },
  ]
}
{% endschema %}
```

Fixes #627

https://github.com/user-attachments/assets/03c1edd0-619a-492c-8085-369410207917

## What's next? Any followup issues?

https://github.com/Shopify/theme-tools/issues/545

## Before you deploy

- [x] I included a minor bump `changeset`

### CODE REVIEW

File: packages/theme-language-server-common/src/json/completions/providers/BlockTypeCompletionProvider.ts [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+export function isBlockTypePath(path: JSONPath) {
+  // We have these cases to support:
+  // - top level blocks.[].type
+  // - presets.[](recursive .blocks.[].type)
+  // - presets.[](recursive .blocks.{}.type)
+  const topLevel = path.at(0);
+  if (topLevel !== 'blocks' && topLevel !== 'presets') return false;
+  if (path.length < 4) return false; // minimum path length
+  const [shouldBeBlocks, _idOrIndex, shouldBeType] = path.slice(-3);
+  return shouldBeBlocks === 'blocks' && shouldBeType === 'type';
```
Comment (charlespwd) [inline packages/theme-language-server-common/src/json/completions/providers/BlockTypeCompletionProvider.ts:82]: Same that was something I used before I planned to split #628 to a different provider

File: packages/theme-language-server-common/src/json/completions/providers/BlockTypeCompletionProvider.ts [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+
+    // Local blocks have their type defined in the schema, there's nothing to complete
+    if (hasLocalBlockDefinitions(schema)) return [];
+
+    const blockNames = await this.getThemeBlockNames(doc.uri);
+
+    return blockNames.map((name) => ({
+      kind: CompletionItemKind.Value,
+      label: `"${name}"`,
+      insertText: `"${name}"`,
```
[the remark this answers is not in the document]
Comment (charlespwd) [reply packages/theme-language-server-common/src/json/completions/providers/BlockTypeCompletionProvider.ts:60]: Unfortunately no, looks like JSON language service wants value completions that start on `"`. Technically a value completion could be a `true | false` so the quotes are added in there to make it so you complete the value as a string.

File: packages/theme-language-server-common/src/json/completions/providers/BlockTypeCompletionProvider.spec.ts [hunk 3]
```diff
@@ -0,0 +1,129 @@
...
+
+      const completions = await jsonLanguageService.completions(params);
+
+      assert(isCompletionList(completions));
+      // @app & @theme come from the JSON schema. There's no way to not include them.
+      expect(completions.items).to.have.lengthOf(blockNames.length + ['@app', '@theme'].length);
+      expect(completions.items.map((item) => item.label)).to.include.members([
+        '"@app"',
+        '"@theme"',
+        ...blockNames.map((x) => `"${x}"`),
```
[the remark this answers is not in the document]
Comment (charlespwd) [reply packages/theme-language-server-common/src/json/completions/providers/BlockTypeCompletionProvider.spec.ts:100]: No no wrapped in `"` :)

File: packages/theme-language-server-common/src/documents/DocumentManager.ts [hunk 4]
```diff
@@ -166,7 +166,11 @@ export class DocumentManager {
 
           /** Lazy and only computed once per file version */
           getSchema: memo(async () => {
-            if (!this.getModeForUri || !this.isValidSchema) return undefined;
+            // For when you don't care about validation... (e.g. tests)
```
[the remark this answers is not in the document]
Comment (charlespwd) [reply packages/theme-language-server-common/src/documents/DocumentManager.ts:169]: That's probably safer yeah. Good call.

File: packages/theme-language-server-common/src/json/completions/providers/BlockTypeCompletionProvider.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (graygilmore) [reply packages/theme-language-server-common/src/json/completions/providers/BlockTypeCompletionProvider.ts:60]: Ooohh I see it now!

**Pull request merged**